1-exercises/B-highlight-words/script.js

```diff
@@ -1,5 +1,50 @@
+// // + Create a `<p>` element.
+// // + Create a `<select>` element.
+// // + Iterate over the options array and create an `<option>` element in the `<select>`
+// for each element.
+// // + You'll need to turn the paragraph into an array to iterate over it.
+// You can use the `.split(" ")` method here.
+// // + Iterate over the array of words.
+// // + For each word, create a `<span>` element and set the innerText to the word,
+// plus a space - " ". Add this to the `<p>`.
+// // + Each `<span>` should have an eventListener that will listen for clicks.
+// // - When clicked, we need to check the value of the `<select>` element using
+// the `.value` property.
+// - We can then update the `background-color` property of the `<span>` with
+//  the value of the select - remember that the value "none" is a special case
+//  and we need to be handled differently.
+// + All of your HTML should go inside the `<div>` with the id **"content"**.
+
+const mainDiv = document.querySelector("#content");
+const pElement = document.createElement("p");
+const selectElement = document.createElement("select");
+const pleaseChoose = document.createElement("option");
+pleaseChoose.innerText = "Please choose a colour";
+selectElement.appendChild(pleaseChoose);
 function highlightWords(paragraph, colours) {
-  // Write your code here...
+  colours.forEach((element) => {
+    let newOptionElement = document.createElement("option");
+    newOptionElement.innerText =
+      element.charAt(0).toUpperCase() + element.slice(1);
+    newOptionElement.value = element;
+    selectElement.appendChild(newOptionElement);
+  });
+  let paragraphArray = paragraph.split(" ");
+  paragraphArray.forEach((element) => {
+    let newSpanElement = document.createElement("span");
+    newSpanElement.innerText = element + " ";
+    pElement.appendChild(newSpanElement);
+    newSpanElement.addEventListener("click", function (event) {
+      event.preventDefault();
+      if (selectElement.value === "none") {
```

Comment:
Instead of using if{}else{} there is also a powerful tool [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator)